### PR TITLE
Add guidelines for rewarding contributions in the context of the TRD grant

### PR DESCRIPTION
**Pull request by amzid**
This PR is in the context of issue #188 concerning the TF grant.
It is a suggestion for the workflow for rewarding contributions and a proper documentation of the payments.

### CODE REVIEW

File: CONTRIBUTORS [hunk 1]
```diff
@@ -1,4 +0,0 @@
-amzid - Thanks for RPC implementation and loding network parameters from tezos node.
```
Comment (jdsika) [inline CONTRIBUTORS:1]: transfer this to the csv in some way to keep history?

File: contributor_payments.csv [hunk 2]
```diff
@@ -0,0 +1 @@
+Tx hash, Tx date, Contributor, Contributor Address, Work effort (hrs), amount in USD, VWAP, amount in XTZ, Reason for payment, Comment
```
Comment (jdsika) [inline contributor_payments.csv:1]: This means we add the compensation for the review as a separate line - I am ok with that, I would add "git commit hashs" with ; or , separators.
Put Contributor and Contributor Address first to keep a read only list of contributors and their permanent xtz addresses at the top of the list. you can transfer the comments from the CONTRIBUTORS file to the comment here

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+name: Pull Request
+about: Create a pull request to contribute to the TRD
+labels: 
+
+---
+IMPORTANT NOTICE: This PR complies with the [guidelines for contribution to the TRD](https://tezos-reward-distributor-organization.github.io/tezos-reward-distributor/contributers.html) and is qualified for being compensated by the TRD grant if approved by the maintainers.
```
Comment (jdsika) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:7]: and MAY qualify for a compensation

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+* **Solution**: If the described issue is a bug report, describe the proposed solution for the reported bug.
+
+* **Implementation**: Rough description/explanation of the implementation choices.
+
+* **Performed tests**: Describe the performed tests and if further tests are needed.
+
+* **Documentation**: Make sure to document the added changes in a proper way (Readme, help section, documentation, comments in code if needed)
+
+**Work effort**: Give your estimate of the work effort in hours. This might be adjusted or discussed by the other contributors in order to keep a fair rewarding process for the efforts.
```
Comment (jdsika) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:21]: add checkboxes at the end to fill out:
[] I extented the documentation
[] I extended the CLI documentation
[] I changed the README file if necessary
[] I added exetended the CI tests for the corresponding feature
[]..... what else you can think of

File: docs/contributers.rst [hunk 5]
```diff
@@ -21,3 +21,14 @@ Principles on the application evolution:
...
+
+Guidelines for rewarding PRs in the context of the TRD grant:
+
+1. Payments are not granted for contributions being paid in the context of another grant or company income.
+2. The contributor should respect the process preceding a PR (start an issue and create a branch) and submit a PR in accordance with the PR template.
+3. The PR can only be merged if approved by at least one maintainer. Before approving a PR, the issue should be fully solved and the changes should be tested and documented.
+4. The PR can only be approved if an agreement about the work effort (in hrs) is made.
+5. The work efforts are paid out in a regular basis. The payments are documented in a dedicated file contributor_payments.csv .
+For each payment, the reason for payment should be documented (commit hash, review efforts, etc.).
+Additionally, the paid amount in USD, the VWAP XTZ-USD on the payment day, the pad amount in XTZ and the transaction hash are documented.
```
Comment (jdsika) [inline docs/contributers.rst:34]: Note that the hours compensated for contributor and reviewer have to be documented in the commit hash of the squash merge into master

**Review by jdsika — CHANGES_REQUESTED** [review]

File: docs/contributers.rst [hunk 6]
```diff
@@ -21,3 +21,14 @@ Principles on the application evolution:
 7. More configuration parameters do not always mean more flexibility. We do not want the application to turn into configuration hell which nobody can use.
 8. The idea of community members is very important because this repo is just a sequence of statements if nobody uses and benefit.
 9. A change will be accepted if it will benefit the community. Particular requests which only very few people will take advantage will not be accepted to leave the code as simple as possible. 
+
+
```
Comment (jdsika) [inline docs/contributers.rst:25]: Add a disclaimer here stating something like:
"this project started in the spirit of free and open source software with voluntary contributions without compensation. You can always donate your contribution free of charge"

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 3 above)
Comment (jdsika) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:7]: "this project started in the spirit of free and open source software with voluntary contributions without compensation. You can always donate your contribution free of charge"

File: contributor_payments.csv (hunk 2 above)
Comment (amzid) [reply contributor_payments.csv:1]: I made an update with the other changes, I will update `contributor_payments.csv` with the history as well once the first payments are made.

**Review by jdsika — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 4 above)
Comment (utdrmac) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:31]: Can we remove the "I" part from the checklist?
[] Extended TravisCI...
[] Extended Sphinx...
.. etc

**Comment by amzid** [comment]
The idea is to add the list of previous contributors to the `contributor_payments.csv` with empty columns except the one of the comment where we can put a list of the highlights of previous contributions. Would that work for you @habanoz @utdrmac?

**Review by jdsika — APPROVED** [review]

**Comment by jdsika** [comment]
@utdrmac final review from your side before I merge?

**Pull request merged**